packages/blade/src/components/Modal/__tests__/Modal.web.test.tsx

```diff
@@ -0,0 +1,244 @@
+  it('renders a basic Modal of large size', () => {
+    const { getByTestId } = renderWithTheme(
+      <Modal isOpen={true} onDismiss={() => {}} accessibilityLabel="Test Modal" size="large">
+        <ModalBody>
+          <Text>Test Content</Text>
+        </ModalBody>
+      </Modal>,
+    );
+    const container = getByTestId('modal-wrapper');
+    expect(container).toMatchSnapshot();
+  });
+
+  it('renders a Modal with Header and Footer', () => {
+    const { getByTestId } = renderWithTheme(
+      <Modal isOpen={true} onDismiss={() => {}} accessibilityLabel="Test Modal">
+        <ModalHeader
+          title="Address Details"
+          subtitle="Saving addresses will improve your checkout experience"
+          trailing={<Badge variant="notice">Action Needed</Badge>}
+          titleSuffix={<Counter intent="positive" value={2} />}
+        />
+        <ModalBody>
+          <Text>Test Content</Text>
+        </ModalBody>
+        <ModalFooter>
+          <Button>Save</Button>
+        </ModalFooter>
+      </Modal>,
+    );
+    const container = getByTestId('modal-wrapper');
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should focus on passed initialFocusRef', () => {
+    const Element = (): React.ReactElement => {
+      const ref = React.useRef(null);
+      return (
+        <Modal
+          isOpen={true}
+          onDismiss={() => {}}
+          accessibilityLabel="Test Modal"
+          size="medium"
+          initialFocusRef={ref}
+        >
+          <ModalBody>
+            <Text>Test Content</Text>
+          </ModalBody>
+          <ModalFooter>
+            <Button ref={ref} accessibilityLabel="save">
+              Save
+            </Button>
+          </ModalFooter>
+        </Modal>
+      );
+    };
+
+    const { getByLabelText } = renderWithTheme(<Element />);
+    expect(getByLabelText('save')).toHaveFocus();
+  });
+
+  it('should handle controlled state of Modal', async () => {
+    const user = userEvents.setup();
+    const onDismiss = jest.fn();
+    const Element = (): React.ReactElement => {
+      const [isOpen, setIsOpen] = React.useState(false);
+      return (
+        <>
+          <Modal
+            isOpen={isOpen}
+            onDismiss={() => {
+              console.log('OOONNNNNDISMISS');
+              onDismiss();
+            }}
+            accessibilityLabel="Test Modal"
+            size="medium"
+          >
+            <ModalHeader title="Modal" />
+            <ModalBody>
+              <Text>Test Content</Text>
+            </ModalBody>
+          </Modal>
+          <Button accessibilityLabel="toggle modal" onClick={() => setIsOpen(true)}>
+            Toggle Modal
+          </Button>
+        </>
+      );
+    };
+
+    const { getByRole, queryByRole } = renderWithTheme(<Element />);
+    await user.click(getByRole('button'));
+    expect(queryByRole('dialog')).toBeInTheDocument();
+    await user.keyboard('{Escape}');
+    expect(onDismiss).toHaveBeenCalledTimes(1);
```

Comment:
You may also expect the dialog to not be there anymore after this point

Reply:
There is some issue with the state not changing on jest randomly which made the test very flaky. Initially I was asserting that only but switched to asserting call of `onDismiss`. If I run the test as `it.only` it passes but if I run all of them together it fails when trying to assert dialog's dismount :face

Reply:
Strange, might be due to animation but non-blocker